### ISSUE TITLE
[Bug] Docker Image Immediately Crashes

### ISSUE THREAD

**Issue by Maxattax97** [coqui-ai/TTS#2068]
### Describe the bug

Docker image immediately crashes

### To Reproduce

1. `docker build .`
2. `docker image ls` to find the has of most recent built image
3. `docker image run ccecb48e2799`

### Expected behavior

Anything that's not an error

### Logs

```shell
Traceback (most recent call last):
  File "/venv/bin/tts", line 5, in <module>
    from TTS.bin.synthesize import main
  File "/root/TTS/bin/synthesize.py", line 11, in <module>
    from TTS.utils.manage import ModelManager
  File "/root/TTS/utils/manage.py", line 11, in <module>
    from TTS.config import load_config
  File "/root/TTS/config/__init__.py", line 10, in <module>
    from TTS.config.shared_configs import *
  File "/root/TTS/config/shared_configs.py", line 5, in <module>
    from trainer import TrainerConfig
  File "/venv/lib/python3.8/site-packages/trainer/__init__.py", line 3, in <module>
    from trainer.model import *
  File "/venv/lib/python3.8/site-packages/trainer/model.py", line 4, in <module>
    import torch
  File "/venv/lib/python3.8/site-packages/torch/__init__.py", line 655, in <module>
    from ._tensor import Tensor
  File "/venv/lib/python3.8/site-packages/torch/_tensor.py", line 15, in <module>
    from torch.overrides import (
  File "/venv/lib/python3.8/site-packages/torch/overrides.py", line 33, in <module>
    from torch._C import (
ImportError: cannot import name '_set_torch_function_mode' from 'torch._C' (/venv/lib/python3.8/site-packages/torch/_C.cpython-38-x86_64-linux-gnu.so)
```

### Environment

```shell
{
    "CUDA": {
        "GPU": [],
        "available": false,
        "version": "10.2"
    },
    "Packages": {
        "PyTorch_debug": false,
        "PyTorch_version": "1.12.1+cu102",
        "TTS": "0.8.0",
        "numpy": "1.22.4"
    },
    "System": {
        "OS": "Linux",
        "architecture": [
            "64bit",
            "ELF"
        ],
        "processor": "",
        "python": "3.10.7",
        "version": "#1 SMP PREEMPT_DYNAMIC Thu, 15 Sep 2022 16:08:26 +0000"
    }
}
```

### Additional context

Sorry this isn't more helpful, but Docker images should be easily reproduced.

**Comment by molekm** [comment]
Same issue here

**Comment by WeberJulian** [comment]
Sorry but I couldn't reproduce your issue.

I tested on two separate envs (one with GPU the other without).
```bash
docker build -t tts .
docker run -it tts --text "Hello."
```

Gives me

```
...
 > Text: hello.
 > Text splitted to sentences.
['hello.']
 > Processing time: 0.21007227897644043
 > Real-time factor: 0.20090621753255167
 > Saving output to tts_output.wav
```

**Comment by erogol** [comment]
Thanks @WeberJulian for checking

**Comment by maroneze** [comment]
I'm having the same issue. I followed [these instructions](https://tts.readthedocs.io/en/latest/docker_images.html):

    docker run --rm --gpus all -v ~/tts-output:/root/tts-output ghcr.io/coqui-ai/tts --text "Hello." --out_path /root/tts-output/hello.wav --use_cuda

And I get the error trace mentioned above.

The same happens if I follow these instructions, from the same page:

```
docker run --rm -it -p 5002:5002 --gpus all --entrypoint /bin/bash ghcr.io/coqui-ai/tts
python3 TTS/server/server.py --list_models #To get the list of available models
python3 TTS/server/server.py --model_name tts_models/en/vctk/vits --use_cuda true
```

Running `python3 TTS/server/server.py` (with any arguments, or none at all) results in the `ImportError` mentioned above.

FWIW, Docker reports the hash of the image as `bc738501c7f9` and "CREATED" as "3 months ago".

With the CPU version (ghcr.io/coqui-ai/tts-cpu), I have no problems at all.

I was able to run `nvidia-smi` through `docker.io/nvidia/cuda:11.8.0-base-ubuntu22.04`, so I believe my GPU is properly setup.

I wonder if there is something system-specific in the way the Docker image is prepared, which might make it fail in some computers but not others. Otherwise, I don't see why a pre-packaged Docker image would fail. I'm willing to help track it down if you believe there is something I could try.
